src/components/external-source/ExternalTypeManager.svelte

```diff
@@ -0,0 +1,1024 @@
+        associatedItems,
+        user,
+      );
+    }
+  }
+
+  function getAssociatedExternalSourcesByEventType(eventType: string | undefined) {
+    if (eventType === undefined) {
+      return [];
+    }
+    const associatedSources = $sourcesUsingExternalEventTypes
+      .filter(entry => entry.types.includes(eventType))
+      .map(entry => entry.key); // NOTE: MAY NEED TO REMOVE THIS - COULD BE A VERY SLOW OPERATION.
+    return associatedSources;
+  }
+
+  function getAssociatedExternalSourcesBySourceType(sourceType: string | undefined) {
+    if (sourceType === undefined) {
+      return [];
+    }
+    let associatedSources = $externalSources.filter(source => source.source_type_name === sourceType);
+    return associatedSources;
+  }
+
+  function getAssociatedDerivationGroupsBySourceTypeName(sourceTypeName: string | undefined) {
+    if (sourceTypeName === undefined) {
+      return [];
+    }
+
+    let associatedDerivationGroups = $derivationGroups.filter(
+      derivationGroup => derivationGroup.source_type_name === sourceTypeName,
+    );
+
+    return associatedDerivationGroups;
+  }
+
+  function hasDeleteDerivationGroupPermissionOnRow(
+    user: User | null,
+    derivationGroup: DerivationGroup | undefined | null,
+  ) {
+    if (derivationGroup === undefined || derivationGroup === null) {
+      return false;
+    } else {
+      return featurePermissions.derivationGroup.canDelete(user, derivationGroup);
+    }
+  }
+
+  function resetUploadForm() {
+    if (fileInput !== null) {
+      fileInput.value = '';
+    }
+    file = undefined;
+    files = undefined;
+    uploadResponseErrors = [];
+    parsedExternalSourceEventTypeSchema = undefined;
+  }
+
+  function onClick() {
+    resetUploadForm();
+  }
+
+  async function handleUpload() {
+    if (files) {
+      file = files[0];
+      if (file !== undefined && /\.json$/.test(file.name)) {
+        uploadResponseErrors = [];
+        const combinedSchema = await parseJSONStream<ExternalSourceEventTypeSchema>(file.stream());
+        await effects.createExternalSourceEventTypes(combinedSchema.event_types, combinedSchema.source_types, user);
+        files = undefined;
+        file = undefined;
+        if (fileInput != null) {
+          fileInput.value = '';
+        }
+        parsedExternalSourceEventTypeSchema = undefined;
+      }
+    }
+  }
+
+  async function parseExternalSourceEventTypeFileStream(stream: ReadableStream) {
+    createExternalSourceEventTypeError.set(null);
+
+    try {
+      parsedExternalSourceEventTypeSchema = await parseJSONStream<ExternalSourceEventTypeSchema>(stream);
+      if (!parsedExternalSourceEventTypeSchema.event_types && !parsedExternalSourceEventTypeSchema.source_types) {
+        parsedExternalSourceEventTypeSchema = undefined;
+        throw new Error('External Source & Event Type Schema has Invalid Format');
+      }
+    } catch (error: any) {
+      createExternalSourceEventTypeError.set(error);
```

Comment:
`error` in this case is an `Error` object, not a `string`. To access the error message to set the store to, you need to do 
```
createExternalSourceEventTypeError.set((error as Error).message);
```